src/extra/others/fragment/lv_fragment.c

```diff
@@ -0,0 +1,392 @@
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return;
+    manager_stack_t * prev = top->prev;
+    bool dialog = top->dialog;
+    if(!dialog && prev) {
+        item_create_fragment(manager, prev, NULL);
+    }
+    item_destroy_obj(manager, top);
+    item_destroy_fragment(top);
+    lv_mem_free(top);
+    if(!dialog && prev) {
+        item_create_obj(manager, prev, manager->container, NULL);
+    }
+    manager->top = prev;
+}
+
+bool lv_fragment_manager_dispatch_event(lv_fragment_manager_t * manager, int which, void * data1, void * data2)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top) return false;
+    lv_fragment_t * instance = top->instance;
+    if(!instance || !top->cls->event_cb) return false;
+    return top->cls->event_cb(instance, which, data1, data2);
+}
+
+#if LV_USE_MSGBOX
+
+void lv_fragment_manager_show(lv_fragment_manager_t * manager, const lv_fragment_class_t * cls, void * args)
+{
+    LV_ASSERT(manager);
+    LV_ASSERT(cls);
+    manager_stack_t * item = item_new(cls);
+    item_create_fragment(manager, item, args);
+    item_create_obj(manager, item, NULL, &lv_msgbox_class);
+    item->dialog = true;
+    manager_stack_t * top = manager->top;
+    item->prev = top;
+    manager->top = item;
+}
+
+#endif
+
+lv_fragment_t * lv_fragment_manager_get_top(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    manager_stack_t * top = manager->top;
+    if(!top)return NULL;
+    return top->instance;
+}
+
+lv_fragment_t * lv_fragment_manager_get_parent(lv_fragment_manager_t * manager)
+{
+    LV_ASSERT(manager);
+    return manager->parent;
+}
+
+lv_fragment_t * lv_fragment_class_create_unmanaged(const lv_fragment_class_t * cls,
+                                                   lv_obj_t * container, void * args)
+{
+    LV_ASSERT(cls);
+    LV_ASSERT(cls->instance_size);
+    LV_ASSERT(cls->create_obj_cb);
+    lv_fragment_t * instance = lv_mem_alloc(cls->instance_size);
+    lv_memset_00(instance, cls->instance_size);
+    instance->cls = cls;
+    if(cls->constructor_cb) {
+        cls->constructor_cb(instance, args);
+    }
+    lv_obj_t * obj = cls->create_obj_cb(instance, container);
+    LV_ASSERT(obj);
+    instance->obj = obj;
+    if(cls->obj_created_cb) {
+        cls->obj_created_cb(instance, obj);
+    }
+    return instance;
+}
+
+void lv_fragment_class_del_unmanaged(lv_fragment_t * fragment)
+{
+    LV_ASSERT(fragment);
+    LV_ASSERT(fragment->obj);
+    LV_ASSERT(!fragment->manager);
+    const lv_fragment_class_t * cls = fragment->cls;
+    if(cls->obj_will_delete_cb) {
+        cls->obj_will_delete_cb(fragment, fragment->obj);
+    }
+    lv_obj_del(fragment->obj);
+    if(cls->obj_deleted_cb) {
+        cls->obj_deleted_cb(fragment, fragment->obj);
+    }
+    if(cls->destructor_cb) {
+        cls->destructor_cb(fragment);
+    }
+    lv_mem_free(fragment);
+}
+
+void lv_fragment_pop(lv_fragment_t * fragment)
```

Comment:
Do we really need this? 
This seems more flexible to me:
```c
lv_fragment_manager_pop(lv_fragment_get_manager(manager));
```

Reply:
Done.